roles/ceph-osd/tasks/docker/start_docker_osd.yml

```diff
@@ -24,46 +35,22 @@
     docker run --net=host \
     --pid=host \
     --privileged=true \
-    --name="ceph-osd-prepare-{{ ansible_hostname }}-dev{{ item.0 | regex_replace('/', '') }}" \
+    --name=ceph-osd-prepare-{{ ansible_hostname }}-dev{{ item.0 | regex_replace('/', '') }} \
     -v /etc/ceph:/etc/ceph \
     -v /var/lib/ceph/:/var/lib/ceph/ \
     -v /dev:/dev \
     -v /etc/localtime:/etc/localtime:ro \
-    -e "OSD_DEVICE={{ item.0 }}" \
-    -e CEPH_DAEMON=OSD_CEPH_DISK_PREPARE \
-    {{ ceph_osd_docker_prepare_env }} \
-    "{{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}"
-  with_together:
-    - "{{ ceph_osd_docker_devices }}"
-    - "{{ osd_prepared.results }}"
-  when:
-    - item.1.get("rc", 0) != 0
-    - ceph_osd_docker_prepare_env is defined
-    - not containerized_deployment_with_kv
-
-- name: prepare ceph osd disk with kv_store
-  shell: |
-    docker run  --net=host   \
-    --pid=host \
-    --privileged=true \
-    --name="ceph-osd-prepare-{{ ansible_hostname }}-dev-{{ item.0 | regex_replace('/', '') }}" \
-    -v /dev:/dev \
-    -v /etc/localtime:/etc/localtime:ro \
-    -e "OSD_DEVICE={{ item.0 }}" \
-    -e "{{ ceph_osd_docker_prepare_env }}" \
+    -e OSD_DEVICE={{ item.0 }} \
+    {{ docker_env_args }} \
     -e CEPH_DAEMON=OSD_CEPH_DISK_PREPARE \
-    -e KV_TYPE={{kv_type}} \
-    -e KV_IP={{kv_endpoint}} \
-    -e KV_PORT={{kv_port}} \
     {{ ceph_osd_docker_prepare_env }} \
-    "{{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}" \
+    {{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}
   with_together:
-    - "{{ ceph_osd_docker_devices }}"
+    - "{{ devices }}"
     - "{{ osd_prepared.results }}"
   when:
-    - item.1.get("rc", 0) != 0
+    - item.2.get("rc", 0) != 0
```

Comment:
Given that there is only devices and osd_prepared.results,  it should be item.1.get() instead of item.2.get()